session/bootstrap.go

```diff
@@ -441,170 +489,17 @@ func upgrade(s Session) {
 		return
 	}
 	// Do upgrade works then update bootstrap version.
-	if ver < version2 {
-		upgradeToVer2(s)
-		ver = version2
-	}
-	if ver < version3 {
-		upgradeToVer3(s)
-	}
-	if ver < version4 {
-		upgradeToVer4(s)
-	}
-
-	if ver < version5 {
-		upgradeToVer5(s)
-	}
-
-	if ver < version6 {
-		upgradeToVer6(s)
-	}
-
-	if ver < version7 {
-		upgradeToVer7(s)
-	}
-
-	if ver < version8 {
-		upgradeToVer8(s)
-	}
-
-	if ver < version9 {
-		upgradeToVer9(s)
-	}
-
-	if ver < version10 {
-		upgradeToVer10(s)
-	}
-
-	if ver < version11 {
-		upgradeToVer11(s)
-	}
-
-	if ver < version12 {
-		upgradeToVer12(s)
-	}
-
-	if ver < version13 {
-		upgradeToVer13(s)
-	}
-
-	if ver < version14 {
-		upgradeToVer14(s)
-	}
-
-	if ver < version15 {
-		upgradeToVer15(s)
-	}
-
-	if ver < version16 {
-		upgradeToVer16(s)
-	}
-
-	if ver < version17 {
-		upgradeToVer17(s)
-	}
-
-	if ver < version18 {
-		upgradeToVer18(s)
-	}
-
-	if ver < version19 {
-		upgradeToVer19(s)
-	}
-
-	if ver < version20 {
-		upgradeToVer20(s)
-	}
-
-	if ver < version21 {
-		upgradeToVer21(s)
-	}
-
-	if ver < version22 {
-		upgradeToVer22(s)
-	}
-
-	if ver < version23 {
-		upgradeToVer23(s)
-	}
-
-	if ver < version24 {
-		upgradeToVer24(s)
-	}
-
-	if ver < version25 {
-		upgradeToVer25(s)
-	}
-
-	if ver < version26 {
-		upgradeToVer26(s)
-	}
-
-	if ver < version27 {
-		upgradeToVer27(s)
-	}
-
-	if ver < version28 {
-		upgradeToVer28(s)
+	for version, upgradeFunc := range bootstrapVersion {
+		if ver < version {
+			upgradeFunc(s)
```

Comment:
should we update to these versions in ascending order from version 2 to version 43?

Reply:
done